techmatters/flex-plugins
# Chi 1040 case routing refactor

Primary reviewer: @GPaoloni 

## Description

* Refactors routing for the editable 'sections' of cases to have a single subroute per section and then a `CaseItemAction` which can be `View`, `Add` or `Edit`
* Leverage the above to implement the 'back to view' requirement of the case edit feature
* Removes usage of activities endpoint, now builds activities from the case information it already has

NOTE: This version 'crosses the Rubicon' in terms of HRM compatibility, the new version is required. Previous PRs would work except for Note / Referral editing with an old version of HRM

TODO: Extra unit tests

### Checklist
- [ X ] Corresponding issue has been opened
- [ X ] Tested for chat contacts
- [ NO ] Tested for call contacts

### Related Issues
Fixes CHI-1040

### Verification steps

Section types: Note, Referral, Document, Perpetrator, Incident

* Edit an item each section type of a case as you are adding the new case as part of a contact.
* Open a case in the case list, add an item each section type then edit it.
* Open a case in the case list with existing items for each section and edit an existing item from each section.
* Repeat the previous 2 steps from the case search

In each case ensure that documents can only have comments edited, and that the document can still be downloaded after editing
